File: CCC00S1.java
import java.util.*;
import java.io.*;
/**
 * CCC '00 S1 - Slot Machines
 * Question type: Simulation
 * 5/5 on DMOJ
 * @author  Tommy Pang
 */
public class CCC00S1 {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;
    public static void main(String[] args) throws IOException {
        int plays = 0;
        int money = Integer.parseInt(br.readLine());
        int a = Integer.parseInt(br.readLine());
        int b = Integer.parseInt(br.readLine());
        int c = Integer.parseInt(br.readLine());
        while (money >= 1) {
            money -= 1;
            plays += 1;
            a += 1;
            if (a == 35) {
                money += 30;
                a = 0;
            }
            if (money == 0) break;
            money -= 1;
            plays += 1;
            b += 1;
            if (b == 100) {
                money += 60;
                b = 0;
            }
            if (money == 0) break;
            money -= 1;
            plays += 1;
            c += 1;
            if (c == 10) {
                money += 9;
                c = 0;
            }
            if (money == 0) break;
        }
        System.out.println("Martha plays " + plays + " times before going broke.");
    }
}

File: CCC00S1.cpp
// Problem: CCC '00 S1 - Slot Machines
// Contest: DMOJ
// URL: https://dmoj.ca/problem/ccc00s1
// Memory Limit: 16 MB
// Time Limit: 1000 ms
// Ryan Liu

#include <bits/stdc++.h>
using namespace std;

int n, a, b, c, sum = 0;

int main() {
  ios_base::sync_with_stdio(0);
  cin.tie(0);
  cout.tie(0);
  cin >> n >> a >> b >> c;

  while (n > 0) {
    if (--n >= 0) {
      a++;
      if (a % 35 == 0) {
        n += 30;
      }
      sum++;
    }
    if (--n >= 0) {
      b++;
      if (b % 100 == 0) {
        n += 60;
      }
      sum++;
    }
    if (--n >= 0) {
      c++;
      if (c % 10 == 0) {
        n += 9;
      }
      sum++;
    }
  }

  cout << "Martha plays " << sum << " times before going broke."
       << "\n";
}
